azurerm/resource_arm_firewall.go

```diff
@@ -297,7 +300,7 @@ func expandArmFirewallIPConfigurations(d *schema.ResourceData) (*[]network.Azure
 	subnetNamesToLock := make([]string, 0)
 	virtualNetworkNamesToLock := make([]string, 0)
 
-	for _, configRaw := range configs {
+	for index, configRaw := range configs {
```

Comment:
we should update this expand method to return an object; such as:

```
type expandedFirewallIPConfigurations struct {
  ipConfigurations []network.AzureFirewallIPConfiguration
  virtualNetworkNames []string
  subnetNames []string
}
```

which would allow us to make this function:

```
func expandArmFirewallIPConfigurations(input []interface{}) (*expandedFirewallIPConfigurations, error) {
```

where the `validate` logic above can be merged into this

Reply:
We are already in `expandArmFirewallIPConfigurations`.